src/runtime_src/core/pcie/linux/device_linux.cpp

```diff
@@ -53,6 +54,106 @@ struct bdf
   }
 };
 
+struct kds_cu_stat
+{
+  using result_type = query::kds_cu_stat::result_type;
+
+  static result_type
+  get(const xrt_core::device* device, key_type)
+  {
+    auto pdev = get_pcidev(device);
+
+    using tokenizer = boost::tokenizer< boost::char_separator<char> >;
+    std::vector<std::string> stats;
+    std::string errmsg;
+
+    // The kds_custat_raw is printing in formatted string of each line
+    // Format: "%d,%s:%s,0x%lx,0x%x,%lu"
+    // Using comma as separator.
+    pdev->sysfs_get("", "kds_custat_raw", errmsg, stats);
+    if (!errmsg.empty())
+      throw std::runtime_error(errmsg);
+
+    result_type cuStats;
+    for (auto& line : stats) {
+      boost::char_separator<char> sep(",");
+      // 5 tokens: index, name, addr, status, usage
+      tokenizer tokens(line, sep);
+      std::string name;
+      uint32_t index = 0;
+      uint64_t base_addr = 0;
+      uint32_t status = 0;
+      uint64_t usage = 0;
```

Comment:
Should these be defined beforehand? If kds_custat_raw is corrupted, it should throw an error (and not print zeros) IMO. And if we decide to throw an error, we can define these from L94 onwards

Reply:
fixed